### ISSUE TITLE
Configure Roles/Permissions & Policies

### ISSUE THREAD

**Issue by nionata** [STEM-C/STEM-C#65]
- [ ] Create roles & permissions for all the endpoints 
   - ie. the public role can only access register/login/etc
- [ ] Create policies to add specific auth logic to key endpoints
   - ie. a mentor should only be able to pull classrooms they belong to

**Comment by nionata** [comment]
@MeghaNagarmunoli brought up a great point that the GET endpoints will return all the data. We need to make sure that we either disable this functionality or have proper roles and policies in place.

**Comment by nionata** [comment]
Helpful policies would be `isStudent` and `isMentor`. This can be used on endpoints like `/sessions/student/activities` to ensure only student users can access. This would make the current logic in that function reusable.

```
        const { student } = ctx.state
        if (!student) return ctx.badRequest(
            'This request is only for student users!',
            { id: 'Session.get.student.activities.user.invalid', error: 'ValidationError' }
        )
```

**Comment by rennemannd** [comment]
I believe this issue has been addressed in #34 and #38 on the frontend and #45 on the backend, recommending this is closed @nionata and @MeghaNagarmunoli

**Comment by nionata** [comment]
> I believe this issue has been addressed in #34 and #38 on the frontend and #45 on the backend, recommending this is closed @nionata and @MeghaNagarmunoli

Good call. The roles and permissions + policies are in place right now. 

Should we remove all the endpoints that no role uses? I'm pretty sure if we just comment them out in `routes.json` then they will be deactivated and won't appear in the permissions selector. The roles should stop their use anyway, but it wouldn't hurt to fully deactivate endpoints that get all records from a content type.

**Comment by rennemannd** [comment]
Good point, I think extra endpoints should be removed or blocked behind some form of authentication. Better safe than sorry with this after all. If a new requirement comes up that happens to use one then it'll be easy to just reinstate it.
